docs/share_usage.md

```diff
@@ -0,0 +1,299 @@
+   it provided that enough space is available on the FPG to accommodate the 
+   share.
+2. If the FPG 'fpg_name' exists and is a legacy FPG, share is created under it
+   provided that enough space is available on the FPG to accommodate the share
+3. If the FPG 'fpg_name' does not exist, it is created with size 
+   'hpe3par_default_fpg_size' configured in hpe.conf provided none of the 3PAR
+    limits are hit. Post FPG creation, share is created under it.
+
+If enough space is not there or any 3PAR limit is hit, the status of share is 
+set to 'FAILED' along with appropriate error message which can be seen while 
+inspecting the share details.
+
+**Note:** ‘size’ can be specified to override the default share size of 1TiB.  
+  
+### Creating a share on a non-default FPG and CPG  
+```  
+docker volume create -d hpe --name <share_name> -o filePersona -o fpg=<fpg_name> -o cpg=<cpg_name>  
+```  
+This command creates a share of default size of 1TiB on the specified FPG ‘fpg_name’. 
+The specified FPG 'fpg_name' may or may not exist.
+
+When this command is executed the plugin does the following:
+1. If the FPG 'fpg_name' exists and it is Docker managed and the specified 
+   CPG 'cpg_name' matches with parent CPG of FPG 'fpg_name', share is created 
+   under it provided that enough space is available on the FPG to accommodate 
+   the share. If specified CPG 'cpg_name' does not match, share creation fails
+   with appropriate error.
+2. If the FPG 'fpg_name' exists and it is a legacy FPG and the specified CPG
+   'cpg_name' matches with the parent CPG of FPG 'fpg_name', share is created 
+   under it provided that enough space is available on the FPG to accommodate
+   the share. If specified CPG 'cpg_name' does not match, share creation fails
+   with appropriate error.
+3. If the FPG 'fpg_name' does not exist, it is created with size 
+   'hpe3par_default_fpg_size' configured in hpe.conf provided none of the 3PAR
+    limits are hit. Post FPG creation, share is created under it.
+
+If enough space is not there or any 3PAR limit is hit, the status of share is 
+set to 'FAILED' along with appropriate error message which can be seen while 
+inspecting the share details.
+
+**Note:** 
+1. ‘size’ can be specified to override the default share size of 1TiB.  
+2. The FPG must have enough capacity to accommodate the share.
+
+### Mounting a share
+```
+docker run -it --rm  --mount src=<share-name>,dst=</mount-dir>,volume-driver=hpe --name <container-name> alpine /bin/sh
+```
+
+This command allows mounting of share 'share-name' inside the container 'container-name' on mount 
+directory 'mount-dir' using alpine image. A share can be mounted multiple times
+on the same host or different hosts that have access to the share. A share that
+is mounted multiple times on a host is unmounted only after the last container 
+mounting it is exited or stopped.
+
+Permissions if present are applied after mounting the share.
+
```

Comment:
**Note:** VFS IPs must be reachable from Docker host for share to be mounted successfully.